src/app/components/homepage/index.js

```diff
@@ -0,0 +1,84 @@
+import Header from 'app/components/homepage/header';
+import Organization from 'app/components/organization-info';
+import React from 'react';
+import UserInfo from 'app/components/user-info';
+import * as Constants from 'app/constants';
+
+const headerTextForUser = {
+  heading: Constants.SHOW_USER_REPO_TEXT,
+  buttonText: Constants.ORG_SEARCH_BUTTON,
+  searchPlaceholder: Constants.USER_SEARCH_PLACEHOLDER,
+}
+
+const headerTextForOrg = {
+  heading: Constants.SHOW_ORG_REPO_TEXT,
+  buttonText: Constants.USER_SEARCH_BUTTON,
+  searchPlaceholder: Constants.ORG_SEARCH_PLACEHOLDER,
+}
+
+class HomePage extends React.Component {
+  constructor(props) {
+    super(props);
+    const { searchName, typeName } = props.match.params;
+    this.state = {
+      searchValue: searchName || '' ,
+      isOrganization: typeName === Constants.USER_TEXT ? false : true,
+    };
+  }
+
+  onChangeHandler = event => {
+    this.setState({ searchValue: event.target.value });
+  };
+
+  onSubmitHandler = event => {
```

Comment:
Move this logic inside header file

Reply:
Moved the above logic to homepage header file from homepage index file.